cmd/gitopsUpdateDeployment.go

```diff
@@ -0,0 +1,176 @@
+package cmd
+
+import (
+	"bytes"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/docker"
+	gitUtil "github.com/SAP/jenkins-library/pkg/git"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"io"
+	"path/filepath"
+)
+
+type gitopsGitUtils interface {
+	CommitSingleFile(filePath, commitMessage string) (plumbing.Hash, error)
+	PushChangesToRepository(username, password string) error
+	PlainClone(username, password, serverURL, directory string) error
+	ChangeBranch(branchName string) error
+}
+
+type gitopsFileUtils interface {
+	TempDir(dir, pattern string) (name string, err error)
+	RemoveAll(path string) error
+}
+
+type gitopsExecRunner interface {
+	RunExecutable(executable string, params ...string) error
+	Stdout(out io.Writer)
+	Stderr(err io.Writer)
+}
+
+type gitUtilsRuntime struct {
+	worktree   *git.Worktree
+	repository *git.Repository
+}
+
+func (g *gitUtilsRuntime) CommitSingleFile(filePath, commitMessage string) (plumbing.Hash, error) {
+	return gitUtil.CommitSingleFile(filePath, commitMessage, g.worktree)
+}
+
+func (g *gitUtilsRuntime) PushChangesToRepository(username, password string) error {
+	return gitUtil.PushChangesToRepository(username, password, g.repository)
+}
+
+func (g *gitUtilsRuntime) PlainClone(username, password, serverURL, directory string) error {
+	var err error
+	g.repository, err = gitUtil.PlainClone(username, password, serverURL, directory)
+	if err != nil {
+		return err
+	}
+	g.worktree, err = g.repository.Worktree()
+	return err
+}
+
+func (g *gitUtilsRuntime) ChangeBranch(branchName string) error {
+	return gitUtil.ChangeBranch(branchName, g.worktree)
+}
+
+func gitopsUpdateDeployment(config gitopsUpdateDeploymentOptions, telemetryData *telemetry.CustomData) {
+	// for command execution use Command
+	var c gitopsExecRunner = &command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runGitopsUpdateDeployment(&config, c, &gitUtilsRuntime{}, piperutils.Files{})
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runGitopsUpdateDeployment(config *gitopsUpdateDeploymentOptions, command gitopsExecRunner, gitopsUpdateDeploymentGitUtilities gitopsGitUtils, gitopsUpdateDeploymentFileUtilities gitopsFileUtils) error {
```

Comment:
You could make the parameter names much shorter. Just "gitUtils" and "fileUtils" seem perfect.